In [1]:
#Method vs function

In [2]:
my_name = "Python"

In [3]:
my_name.upper()

'PYTHON'

In [4]:
#Functions
#def = define

In [6]:
def hello_python():
    print("hello")
    print("python")

In [8]:
hello_python()

hello
python


Pythonissa "def" -avainsanaa käytetään funktion määrittämiseen. Funktiot ovat uudelleenkäytettäviä koodilohkoja, jotka suorittavat tiettyjä tehtäviä. Ne auttavat koodin järjestämisessä, lukemisen helpottamisessa ja toistojen vähentämisessä.

Funktion määrittämiseksi Pythonissa aloitetaan "def" -avainsanalla, jota seuraa funktion nimi ja sulkeet (). Tarvittavat parametrit tai argumentit sijoitetaan näiden sulkeiden sisään. Funktion määritelmää seuraa kaksoispiste :.

Tässä on yleinen syntaksi funktion määrittämiseen Pythonissa:

def funktion_nimi(parametri1, parametri2, ...):
#Funktion runko
#Suoritettava koodi
#Valinnaisesti palautuslause
Funktion rungossa kirjoitat koodin, joka määrittelee funktion toiminnallisuuden. Tähän voi kuulua mikä tahansa kelvollinen Python-koodi, kuten muuttujien määrittelyjä, ehtolauseita, silmukoita ja muita funktiokutsuja.

Valinnaisesti funktiolla voi olla palautuslause, joka määrittelee arvon tai arvot, jotka palautetaan, kun funktiota kutsutaan. Jos palautuslauseita ei anneta, funktio palauttaa oletusarvoisesti None-arvon.

Funktiota käytetään yksinkertaisesti kutsumalla sitä sen nimellä, jota seuraa sulkeet, ja välittämällä tarvittavat argumentit. Funktio suoritetaan, ja jos siitä palautuu arvo, se voidaan asettaa muuttujaan tai käyttää halutulla tavalla.

In [10]:
def hello_name(name):
    print("hello")
    print(name)

In [11]:
hello_name("Python")

hello
Python


In [12]:
hello_name("JavaScript")

hello
JavaScript


In [14]:
def sum_example(num1, num2):
    print(num1+num2)

In [15]:
sum_example(5,10)

15


In [16]:
#return

In [17]:
def summation(num1,num2,num3):
    print(num1+num2+num3)

In [18]:
summation(10,2,8)

20


In [19]:
 def control_string(s):
        if s[0] == "P":
            print("P")

In [20]:
control_string("Python")

P


In [21]:
def args_sum(*args):
    return sum(args)

Tämä koodi määrittelee funktion nimeltä args_sum, joka ottaa vastaan mielivaltaisen määrän argumentteja. Tähden (*) edessä oleva args ilmaisee, että funktio vastaanottaa muuttuvan määrän argumentteja ja tallentaa ne tupleen nimeltä args.

Funktio käyttää sitten sum()-funktiota laskemaan args-tuplen kaikkien arvojen summan. Lopuksi summa palautetaan funktion tulosteena.

In [22]:
args_sum(20,30,40,90,872,12)

1064

Tässä rivillä funktio args_sum kutsutaan argumenteilla 20, 30, 40, 90, 872, 12. Nämä arvot välitetään erillisinä argumentteina, ja funktio laskee niiden summan.

Joten tämän koodin tuloste olisi 1064, mikä on kaikkien args_sum-funktiolle välitettyjen arvojen summa.

In [25]:
def kwargs_example(**kwargs):
    print(kwargs)

Tämä koodi määrittelee kwargs_example nimisen funktion. Tämä funktio hyväksyy muuttuvan määrän avain-arvo -pareja. Kaksi tähteä (**) kwargs ilmaisee, että funktio vastaanottaa nimettömän sanakirjan (dictionary), joka sisältää nämä avain-arvo -parit.

Funktio tulostaa napattujen avain-arvo -parien sisällön näytölle (print(kwargs)).

In [33]:
kwargs_example(apple =100, banana= 150,orange =250)

Apleee


Tässä rivillä kwargs_example funktiota kutsutaan avain-arvo -pareilla apple=100, banana=150, orange=250. Nämä arvot välitetään erillisinä argumentteina ja funktio napataan ne.

In [34]:
def kwargs_example(**kwargs):
    if "apple" in kwargs:
        print("Apleee")
    else:
        print("Sorry")

In [37]:
kwargs_example(banana= 150,orange =250)

Sorry
